synedrion/src/uint/traits.rs

```diff
@@ -27,289 +14,48 @@ pub(crate) const fn upcast_uint<const N1: usize, const N2: usize>(value: Uint<N1
     Uint::from_words(result_words)
 }
 
-pub trait UintLike:
-    Integer
-    + Encoding
-    + Hashable
-    + RandomPrimeWithRng
-    + RandomMod
-    + Random
-    + subtle::ConditionallySelectable
-{
-    type ModUint: UintModLike<RawUint = Self>;
-    fn from_xof(reader: &mut impl XofReader, modulus: &NonZero<Self>) -> Self;
-    fn add_mod(&self, rhs: &Self, modulus: &NonZero<Self>) -> Self;
-    fn sub_mod(&self, rhs: &Self, modulus: &NonZero<Self>) -> Self;
-    fn trailing_zeros(&self) -> usize;
-    fn inv_mod(&self, modulus: &Self) -> CtOption<Self>;
-    fn wrapping_sub(&self, other: &Self) -> Self;
-    fn wrapping_mul(&self, other: &Self) -> Self;
-    fn wrapping_add(&self, other: &Self) -> Self;
-    fn bits(&self) -> usize;
-    fn bits_vartime(&self) -> usize;
-    fn bit(&self, index: usize) -> Choice;
-    fn bit_vartime(&self, index: usize) -> bool;
-    fn neg(&self) -> Self;
-    fn neg_mod(&self, modulus: &Self) -> Self;
-    fn shl_vartime(&self, shift: usize) -> Self;
-    fn shr_vartime(&self, shift: usize) -> Self;
-    fn to_mod(&self, precomputed: &<Self::ModUint as UintModLike>::Precomputed) -> Self::ModUint {
-        Self::ModUint::new(self, precomputed)
+pub trait ToMod: Integer {
```

Comment:
Do we really need a custom trait here? Can we just use `crypto_bigint::Monty`?

Reply:
This is a convenience because type inference is being wonky. Take this example with the trait:

```rust
        let inv_totient = totient
            .as_ref()
            .to_montgomery(public_key.precomputed_modulus())
            .invert()
            .expect("Bla bla");

```

…and without the trait:

```rust
        let inv_totient = <<P as PaillierParams>::UintMod as Monty>::new(
            totient.into_inner(),
            <P::UintMod as Monty>::Params::clone(public_key.precomputed_modulus()),
        )
        .invert()
        .expect("Bla bla");

```

Certainly doable, but readability suffers!